tests/end_to_end/smoke_test.py

```diff
@@ -5,40 +5,44 @@
 import os
 
 SCRIPT = """
-source {envloc}/bin/activate;
+source {envloc}/{folder}/activate;
+python ok
 python ok {args} > {stdoutloc} 2> {stderrloc}
 """
 
 class SmokeTest(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
-        cls.clean_env_loc = tempfile.TemporaryDirectory().name
+        cls.clean_env_loc = tempfile.TemporaryDirectory().name.replace("\\", "/")
         cls.create_clean_env()
 
     @classmethod
     def create_clean_env(cls):
-        subprocess.check_call(["virtualenv", "-q", "-p", "python", cls.clean_env_loc])
+        subprocess.check_call(["virtualenv", "-p", "python", cls.clean_env_loc])
 
     def setUp(self):
-        self.directory = tempfile.TemporaryDirectory().name
+        self.directory = tempfile.TemporaryDirectory().name.replace("\\", "/")
         publish.package_client(self.directory)
 
     def run_ok(self, *args):
-        out_loc = tempfile.NamedTemporaryFile().name
-        err_loc = tempfile.NamedTemporaryFile().name
+        _, out_loc = tempfile.mkstemp()
```

Comment:
why `mkstemp`?

Note that the temporary file/directory objects must be kept alive as long as those entities are needed. If the issue you're seeing is that they're ceasing to exist, that's why.

Reply:
mkstemp is the way to make temporary files in python3. mktemp is deprecated

Reply:
I think this should be fine: "Unlike TemporaryFile(), the user of mkstemp() is responsible for deleting the temporary file when done with it."

I'll use mkdtemp() for the directory for consistency